### DISCUSSION TITLE
Action groups

### DISCUSSION THREAD

**Discussion by rishabkaushik** [codefori/vscode-ibmi#1834] [Ideas]
It will be helpful if we can combine multiple actions and use the completion status to execute another action based on it. 
e.g. 

1. Running a compile command, if it is successful, call the program. 
2. Run a compile module command, if it is successful, create a program object for it.

**Comment by chrjorgensen** [comment]
@rishabkaushik You know you can put multiple commands in a single action, right? Just have one or more newlines between the commands:

![image](https://github.com/codefori/vscode-ibmi/assets/13275072/143c3385-ad01-4316-a6bf-ff6b378abdb8)

You can even prompt the commands by adding a `?` to the front of the commands you want prompted...

The commands will be run one after the other as long as the command does not fail... and I guess this is what you're looking for?

**Reply by rishabkaushik** [reply]
Hey @chrjorgensen,
 
Ah! I did not know you can run multiple commands like that though I was aware of prompting. Thanks!

One more thing, what if I want to execute an action like opening a source member for edit? The use case I have is that, we have a source control mechanism - which has a command to copy the member before editing to another library say STAGED. I have defined a custom action for the command and it works fine and copies the member to STAGED library. I do have a filter to browse and edit that member and I have to manually open the member.

What I want to achieve is, once I run the command it copies the member to STAGED lib, then at VS Code the newly created member is opened without me navigating to the STAGED library. Is it a viable option to have?

**Reply by chrjorgensen** [reply]
@rishabkaushik Sorry, but it is not possible to send commands to Code for IBM i from the actions. Actions will always send the command to the server, not Code for IBM i. Your request would require some coding in the extension to make that possible... 😞 

Your idea has been noted and will be considered by the team - but you're also welcome to clone the source code and make the change yourself, if you have the skill and code TypeScript, and then send your changes in a PR. 😃

**Reply by rishabkaushik** [reply]
@chrjorgensen 
That is some motivation to learn something new! I'll hop on to the documentation this weekend. Thanks.
